lib/drivers/weitai/weiTaiUSB.js

```diff
@@ -0,0 +1,1683 @@
+          );
+          debug('Received', _.toUpper(incoming.toString('hex')));
+          if (count == 1) {
+            const command = WeitaiUSB.getCommand(incomingA);
+            const commandBody = new Buffer(unpackFrame(command));
+            commandBodyLength = commandBody.readInt32LE(4);
+            count++;
+          } else {
+            commandBodyLength = commandBodyLength - incoming.length;
+            //  console.log("transincoming length", incoming.length);
+            if (commandBodyLength == 0) {
+              //读取完毕
+              //  console.log("getRes end");
+              done = true;
+              pdaSn = WeitaiUSB.parseSettingPacket(incomingA, 'PDASN', cb);
+            }
+          }
+        });
+    }
+    return pdaSn;
+  }
+
+  async getConfig(data, cb) {
+    //start
+    //  const buffer = Buffer.alloc(28);
+    //setting读取
+    //低血糖目标
+    this.current = 0;
+    let settings = {};
+    try {
+      for (let i = 0; i < this.setTypes.length; i++) {
+        settings = await this.getSetting(cb);
+      }
+    } catch (e) {
+      return cb({ code: 'E_READ_FILE' }, null);
+    }
+    let pdaSn = '';
+    try {
+      pdaSn = await this.getPdaSn(cb);
+    } catch (e) {
+      return cb({ code: 'E_READ_FILE' }, null);
+    }
+    // debug('settings', settings);
+    let done = false;
+    let count = 1;
+    let commandBodyLength = 0;
+    const buffer = WeitaiUSB.buildPacket(Buffer.alloc(0));
+    await this.usbDevice.transferOut(this.outEndpoint.address, buffer);
+    var incomingA = Buffer.alloc(0);
+    while (!done) {
+      await this.usbDevice
+        .transferIn(this.inEndpoint.address, 10240)
+        .then((res) => {
+          const incoming = res;
+          console.log(incoming.length);
+          incomingA = Buffer.concat(
+            [incomingA, incoming],
+            incomingA.length + incoming.length
+          );
+          debug('Received', _.toUpper(incoming.toString('hex')));
+          if (count == 1) {
+            const command = WeitaiUSB.getCommand(incomingA);
+            const commandBody = new Buffer(unpackFrame(command));
+            commandBodyLength = commandBody.readInt32LE(4);
+            count++;
+          } else {
+            commandBodyLength = commandBodyLength - incoming.length;
+            //  console.log("transincoming length", incoming.length);
+            if (commandBodyLength == 0) {
+              //读取完毕
+              //  console.log("getRes end");
+              done = true;
+            }
+          }
+        });
+    }
+    const getMostRecentUpload = promisify(
+      this.cfg.api.getMostRecentUploadRecord
+    );
+
+    //chuliSN
+    this.cfg.lastUpload = 0;
+    const inComeRes = WeitaiUSB.parsePacket(incomingA, this.cfg, cb); //inComeRes:{sn:0}
+    this.cfg.deviceInfo.deviceId = 'equil-' + pdaSn;
+    this.cfg.deviceInfo.serialNumber = pdaSn;
+    const res = await getMostRecentUpload(
+      this.cfg.groupId,
+      'equil-' + pdaSn
```

Comment:
```suggestion
     this.cfg.deviceInfo.deviceId
```

Reply:
> 

@gniezen  Thank you for your reply. I will submit it as soon as possible